DTAConfig/CustomSettings/FileSettingCheckBox.cs

```diff
@@ -0,0 +1,100 @@
+    /// </summary>
+    public class FileSettingCheckBox : XNAClientCheckBox, ICustomSetting
+    {
+        public FileSettingCheckBox(WindowManager windowManager) : base(windowManager) { }
+
+        public FileSettingCheckBox(WindowManager windowManager,
+            string sourceFilePath, string destinationFilePath, FileOperationOptions options,
+            bool reversed) : base(windowManager)
+        {
+            files.Add(new FileSourceDestinationInfo(sourceFilePath, destinationFilePath, options));
+            this.reversed = reversed;
+        }
+
+        private List<FileSourceDestinationInfo> files = new List<FileSourceDestinationInfo>();
+        private bool reversed;
+        private bool originalState;
+        private bool restartRequired;
+
+        public override void GetAttributes(IniFile iniFile)
+        {
+            base.GetAttributes(iniFile);
+
+            var section = iniFile.GetSection(Name);
+
+            if (section == null)
+                return;
+
+            int i = 0;
+            while (true)
+            {
+                string fileInfo = section.GetStringValue($"File{i}", string.Empty);
+
+                if (string.IsNullOrWhiteSpace(fileInfo))
+                    break;
+
+                string[] parts = fileInfo.Split(',');
+                if (parts.Length < 2)
+                {
+                    Logger.Log($"Invalid FileSettingCheckBox information in {Name}: {fileInfo}");
+                    continue;
+                }
+
+                FileOperationOptions options = default;
+                if (parts.Length >= 3)
+                    Enum.TryParse(parts[2], out options);
+
+                files.Add(new FileSourceDestinationInfo(parts[0], parts[1], options));
+
+                i++;
+            }
+        }
+
+        public override void ParseAttributeFromINI(IniFile iniFile, string key, string value)
+        {
+            switch (key)
+            {
+                case "Reversed":
+                    reversed = Conversions.BooleanFromString(value, false);
+                    return;
+                case "RestartRequired":
+                    restartRequired = Conversions.BooleanFromString(value, false);
+                    return;
+            }
+
+            base.ParseAttributeFromINI(iniFile, key, value);
+        }
+
+        public void Load()
+        {
+            Checked = reversed != File.Exists(ProgramConstants.GamePath + files[0].DestinationPath);
+            originalState = Checked;
+        }
+
+        public bool RefreshSetting()
+            // TODO implement custom logic for refreshing the checkbox
+            => false;
+
+        public bool Save()
+        {
+            if (reversed != Checked)
+                files.ForEach(f => f.Apply());
+            else
+                files.ForEach(f => f.Revert());
+
+            return restartRequired && (Checked != originalState);
+        }
+    }
+}
```

Comment:
To prevent duplicate code, this class should be implemented as a `CustomSettingFileCheckBox`. Maybe implementing this as a derived class would be easy.

Is there a reason why the new class is called `CustomSettingFileCheckBox` instead of following the naming convention of this class with `CustomFileSettingCheckBox` btw?

Reply:
> Is there a reason why the new class is called `CustomSettingFileCheckBox` instead of following the naming convention of this class with `CustomFileSettingCheckBox` btw?

Yes, the interface for all custom settings is named as `ICustomSetting` so I followed that and prefixed the class name like so because there could be non-file settings and because of that I thought that it would make more sense to operate with custom settings, not settings that allow for custom files.

Reply:
Also this won't be possible without breaking compatibility with older builds and/or user configurations.

Reply:
Latest implementation of `FileSettingCheckBox` achieves full backwards compatibility with the old client configuration whilst using a minimal amount of duplicate or bespoke code & meshing in with the new / altered code in `XNAOptionsPanel`. There is room for improvement, such as looking into the possibility of implementing `FileSettingsCheckBox` as a class inherited from `CustomSettingFileCheckBox`, but there's probably further discussion warranted for the aim of such move (is the aim to keep 1:1 backwards compatibility or offer a transitional system to the new INI setting-based system?) that, along with any further code changes regarding this, are probably best deferred to a later date and outside this PR.